src/rcc.rs

```diff
@@ -306,8 +306,13 @@ impl CFGR {
             })
             .unwrap_or(rcc::cfgr::HPRE_A::DIV1);
 
-        // TODO is this casting safe?
-        let hclk = sysclk / (1 << (hpre_div as u32));
+        let hpre_div_value = if u8::from(hpre_div) != 0 {
+            u8::from(hpre_div) - 7
+        } else {
+            0
+        };
```

Comment:
May I suggest using [`TryFrom`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) / [`TryInto`](https://doc.rust-lang.org/std/convert/trait.TryInto.html)? This could be used for the other cases below as well.

Example ([link to playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2d95af577129c5c5cc3dbd2ac4a373a3)):
```rust
use std::convert::TryInto;

fn number_into_minus_seven<T: TryInto<u8>>(number: T) -> u8 {
    number.try_into().map(|x| x - 7).unwrap_or_default()
}

fn main() {
    assert_eq!(number_into_minus_seven(255), 248);
    assert_eq!(number_into_minus_seven(256), 0);
}
```

Reply:
Thank you for the suggestion!

Sadly, this won't help. As the conversion from the enum to `u8` is not the
problem. The problem lies into subtracting the offset.

If I  understand it correctly, the conversion should always succeed, therefor
`try_from` is not needed.

I use `checked_sub` now. I could apply the tip using `unwrap_or_default`. Thank
you for that :)

Reply:
Looks nice and clean :+1: